docs/developer/development-environment.md

```diff
@@ -0,0 +1,130 @@
+# Configuring Development Environment
+
+This article is written to guide developers to create development environment for developers. This guide can also be used to build Fed-BioMed software from its source.
+
+
+This article will guide you to:
+
+- clone Fed-BioMed source repository.
+- setup a recommended tool to manage different Python versions.
+- setup some optional tools to manage isolated python environments such as `conda` and `virtualenv`.
+- setup pdm to manage dependencies and build operations, and it usage.
+
+
+## Clone the repository
+The first step is to clone the repository please go to (Fed-BioMed GitHub Repository)[https://github.com/fedbiomed/fedbiomed] and clone it your local environment.
+
+```
+git clone https://github.com/fedbiomed/fedbiomed <path-to-clone>
+cd <path-to-clone>
+```
+
+
+## Compatible Python version
+
+Before start using Fed-BioMed or start developing on Fed-BioMed please make sure that required Python version is installed on your machine. Please go to `pyrpoject.toml` file located in the root of Fed-BioMed clone, and check required Python version interval.
+
+!!! note "Tool for different Python version"
+
+    You can use the tool `pyenv` to be able to instal specific Python versions into your local environment
```

Comment:
redirect to https://github.com/pyenv/pyenv?tab=readme-ov-file#1-automatic-installer-recommended for pyenv installation